pkg/simulator/filesystem.go

```diff
@@ -30,13 +33,27 @@ func (fs *FileSystemManager) BootstrapDirectories() error {
 		fs.createdDirs = append(fs.createdDirs, createdInstancePath)
 	}
 
-	createdFirmwarePath, err := mkdirAll(fs.firmwareDir, 0755)
+	createdFirmwareDirPathFileFolder, err := mkfile(fs.firmwareDirPathFile, 0755)
+	if err != nil {
+		fs.Cleanup()
+		return fmt.Errorf("failed to create firmware path directory: %w", err)
+	}
+	if createdFirmwareDirPathFileFolder != "" {
+		fs.createdDirs = append(fs.createdDirs, createdFirmwareDirPathFileFolder)
+	}
+	// Write Firmware path
+	if err := os.WriteFile(fs.firmwareDirPathFile, []byte(fs.firmwareDir), 0644); err != nil {
+		fs.Cleanup()
+		return fmt.Errorf("failed to write firmware path file: %w", err)
+	}
+
+	createdFirmwareDir, err := mkdirAll(fs.firmwareDir, 0755)
 	if err != nil {
 		fs.Cleanup()
 		return fmt.Errorf("failed to create firmware directory: %w", err)
 	}
-	if createdFirmwarePath != "" {
-		fs.createdDirs = append(fs.createdDirs, createdFirmwarePath)
+	if createdFirmwareDir != "" {
```

Comment:
I see this pattern is already in the code, but I'm a bit confused by it - I can't see why it would ever be nil and it feels like it would be an error condition